ddtrace/tracer/tracer_test.go

```diff
@@ -89,10 +89,13 @@ func TestTracerCleanStop(t *testing.T) {
 		}()
 	}
 
+	resetLogWriter := setLogWriter(ioutil.Discard)
+	defer resetLogWriter()
```

Comment:
Not sure if the lengthy return value is needed. One line could do:
```suggestion
	defer setLogWriter(ioutil.Discard)()
```

Reply:
I was considering to leave the same comment initially, but then decided the variable name helps to clarify what's going on. Oh well, the new version is fine too, so no worries : )